src/adapter/src/client.rs

```diff
@@ -363,6 +368,44 @@ impl SessionClient {
         self.session = Some(res.session);
         res.result
     }
+
+    pub fn add_idle_in_transaction_session_timeout(&mut self) {
+        let session = self.session();
+        let timeout_dur = session.vars().idle_in_transaction_session_timeout();
+        if !timeout_dur.is_zero() {
+            let timeout_dur = timeout_dur.clone();
+            if let Some(txn) = session.transaction().inner() {
+                let txn_id = txn.id.clone();
+                let timeout = TimeoutType::IdleInTransactionSession(txn_id);
+                self.timeouts.add_timeout(timeout, timeout_dur);
+            }
+        }
+    }
+
+    pub fn remove_idle_in_transaction_session_timeout(&mut self) {
+        let session = self.session();
+        if let Some(txn) = session.transaction().inner() {
+            let txn_id = txn.id.clone();
+            self.timeouts
+                .remove_timeout(&TimeoutType::IdleInTransactionSession(txn_id));
+        }
+    }
+
+    pub async fn recv_timeout(&mut self) -> Option<TimeoutType> {
+        self.timeouts.recv().await
+    }
+
+    pub fn handle_timeout(&mut self, timeout: TimeoutType) -> Result<(), AdapterError> {
+        if self.timeouts.remove_timeout(&timeout) {
+            match timeout {
+                TimeoutType::IdleInTransactionSession(_) => {
```

Comment:
Should the txn id be checked here? Otherwise I think a previous txn's timeout could trigger this?